In [22]:
import pandas as pd
import numpy as np

In [23]:
filePath = '../thailand_foods.csv'

In [24]:
df = pd.read_csv(filePath)

In [25]:
df

en_name            th_name  \
0                  Koi               ก้อย   
1            Kuai Chap            กวยจั๊บ   
2            Kuai Tiao         ก๋วยเตี๋ยว   
3      Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4    Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
..                 ...                ...   
320              A Kao             อาเกาะ   
321              A Law           อาละหว่า   
322              A Law              อาลัว   
323              O Aew            โอ้เอ๋ว   
324      Ice Cream Pad           ไอติมผัด   

                                          ingredients     course  \
0              Beef+lime juice+fish sauce+chili+herbs  main dish   
1            Pork belly+rice noodles+soy sauce+garlic  main dish   
2    Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3              Beef+noodles+curry powder+coconut milk  main dish   
4                    Chicken+noodles+soy sauce+garlic  main dish   
..                                                ...        ...   
320                                           Unknown    dessert   
321                                           Unknown    dessert   
322                                           Unknown    dessert   
323                                           Unknown    dessert   
324                                           Unknown    dessert   

             province     region  
0    Ubon Ratchathani  Northeast  
1             Bangkok    Central  
2             Bangkok    Central  
3           Ayutthaya    Central  
4             Bangkok    Central  
..                ...        ...  
320           Unknown    Unknown  
321           Unknown    Unknown  
322           Unknown    Unknown  
323           Unknown    Unknown  
324           Unknown    Unknown  

[325 rows x 6 columns]

In [26]:
df.shape

(325, 6)

In [27]:
df.head(10)

en_name            th_name  \
0                      Koi               ก้อย   
1                Kuai Chap            กวยจั๊บ   
2                Kuai Tiao         ก๋วยเตี๋ยว   
3          Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4        Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
5         Kuai Tiao Pak Mo  ก๋วยเตี๋ยวปากหม้อ   
6            Kuai Tiao Mon      ก๋วยเตี๋ยวมอญ   
7           Kuai Tiao Ruea     ก๋วยเตี๋ยวเรือ   
8        Kung Chae Nam Pla      กุ้งแช่น้ำปลา   
9  Koey Yeung Kok Siang Bi  เกยย้งก๊กเซียงบี๊   

                                        ingredients     course  \
0            Beef+lime juice+fish sauce+chili+herbs  main dish   
1          Pork belly+rice noodles+soy sauce+garlic  main dish   
2  Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3            Beef+noodles+curry powder+coconut milk  main dish   
4                  Chicken+noodles+soy sauce+garlic  main dish   
5                      Pork+herbs+rice flour+garlic  main dish   
6                Rice noodles+pork+fish sauce+herbs  main dish   
7        Rice noodles+pork+blood broth+bean sprouts  main dish   
8                    Shrimp+fish sauce+garlic+chili  main dish   
9                              Herbs+pork+soy sauce  main dish   

            province     region  
0   Ubon Ratchathani  Northeast  
1            Bangkok    Central  
2            Bangkok    Central  
3          Ayutthaya    Central  
4            Bangkok    Central  
5      Nakhon Pathom    Central  
6         Nonthaburi    Central  
7          Ayutthaya    Central  
8  Coastal provinces      South  
9            Bangkok    Central

In [28]:
df.tail(10)

en_name        th_name                    ingredients   course  \
315        Luk Chup         ลูกชุบ  Mung beans+coconut milk+sugar  dessert   
316        Som Chun         ส้มฉุน         Herbs+fruit+fish sauce    salad   
317   Sankhaya Kuan   สังขยา (กวน)         Flour+coconut milk+egg  dessert   
318  Sankhaya Nueng  สังขยา (นึ่ง)         Flour+coconut milk+egg  dessert   
319          Ang Ku         อังกู๊                        Unknown  dessert   
320           A Kao         อาเกาะ                        Unknown  dessert   
321           A Law       อาละหว่า                        Unknown  dessert   
322           A Law          อาลัว                        Unknown  dessert   
323           O Aew        โอ้เอ๋ว                        Unknown  dessert   
324   Ice Cream Pad       ไอติมผัด                        Unknown  dessert   

    province   region  
315  Various  Various  
316  Various  Various  
317  Various  Various  
318  Various  Various  
319  Unknown  Unknown  
320  Unknown  Unknown  
321  Unknown  Unknown  
322  Unknown  Unknown  
323  Unknown  Unknown  
324  Unknown  Unknown

In [29]:
unknown_count = (df['ingredients'] == 'Unknown').sum()

In [30]:
print("Total rows with Unknown ingredients:", unknown_count)

Total rows with Unknown ingredients: 100


In [36]:
thaiFood.to_csv("thaiDish.csv", index=False)

In [37]:
df = pd.read_csv('thaiDish.csv')

In [38]:
df

en_name                                       ingredients  \
0                  Koi            Beef+lime juice+fish sauce+chili+herbs   
1            Kuai Chap          Pork belly+rice noodles+soy sauce+garlic   
2            Kuai Tiao  Rice noodles+pork+bean sprouts+garlic+fish sauce   
3      Kuai Tiao Khaek            Beef+noodles+curry powder+coconut milk   
4    Kuai Tiao Kua Kai                  Chicken+noodles+soy sauce+garlic   
..                 ...                                               ...   
320              A Kao                                           Unknown   
321              A Law                                           Unknown   
322              A Law                                           Unknown   
323              O Aew                                           Unknown   
324      Ice Cream Pad                                           Unknown   

        course          province     region  
0    main dish  Ubon Ratchathani  Northeast  
1    main dish           Bangkok    Central  
2    main dish           Bangkok    Central  
3    main dish         Ayutthaya    Central  
4    main dish           Bangkok    Central  
..         ...               ...        ...  
320    dessert           Unknown    Unknown  
321    dessert           Unknown    Unknown  
322    dessert           Unknown    Unknown  
323    dessert           Unknown    Unknown  
324    dessert           Unknown    Unknown  

[325 rows x 5 columns]

In [41]:
mainDishCount = (df['course'] == 'main dish').sum()

In [42]:
print('totalOfMainDish', mainDishCount)

totalOfMainDish 173


In [43]:
dessertCount = (df['course'] == 'dessert').sum()

In [46]:
print('totalDessertCount', dessertCount)

totalDessertCount 70


In [47]:
snackCount = (df['course'] == 'snack').sum()

In [48]:
print('totalSnackCount', snackCount)

totalSnackCount 79


Total Of MainDish = 173,  Total of Dessert = 70, Total of Snack = 79

In [49]:
duplicate_count = df.duplicated().sum()
print("Total duplicate rows:", duplicate_count)

Total duplicate rows: 1


In [50]:
df_no_duplicates = df.drop_duplicates()

In [51]:
df

en_name                                       ingredients  \
0                  Koi            Beef+lime juice+fish sauce+chili+herbs   
1            Kuai Chap          Pork belly+rice noodles+soy sauce+garlic   
2            Kuai Tiao  Rice noodles+pork+bean sprouts+garlic+fish sauce   
3      Kuai Tiao Khaek            Beef+noodles+curry powder+coconut milk   
4    Kuai Tiao Kua Kai                  Chicken+noodles+soy sauce+garlic   
..                 ...                                               ...   
320              A Kao                                           Unknown   
321              A Law                                           Unknown   
322              A Law                                           Unknown   
323              O Aew                                           Unknown   
324      Ice Cream Pad                                           Unknown   

        course          province     region  
0    main dish  Ubon Ratchathani  Northeast  
1    main dish           Bangkok    Central  
2    main dish           Bangkok    Central  
3    main dish         Ayutthaya    Central  
4    main dish           Bangkok    Central  
..         ...               ...        ...  
320    dessert           Unknown    Unknown  
321    dessert           Unknown    Unknown  
322    dessert           Unknown    Unknown  
323    dessert           Unknown    Unknown  
324    dessert           Unknown    Unknown  

[325 rows x 5 columns]

Remove Duplicate Value

In [52]:
duplicates = df[df.duplicated()]
print(duplicates)

    en_name ingredients   course province   region
322   A Law     Unknown  dessert  Unknown  Unknown


In [53]:
df = df.drop(322)

In [54]:
df

en_name                                       ingredients  \
0                  Koi            Beef+lime juice+fish sauce+chili+herbs   
1            Kuai Chap          Pork belly+rice noodles+soy sauce+garlic   
2            Kuai Tiao  Rice noodles+pork+bean sprouts+garlic+fish sauce   
3      Kuai Tiao Khaek            Beef+noodles+curry powder+coconut milk   
4    Kuai Tiao Kua Kai                  Chicken+noodles+soy sauce+garlic   
..                 ...                                               ...   
319             Ang Ku                                           Unknown   
320              A Kao                                           Unknown   
321              A Law                                           Unknown   
323              O Aew                                           Unknown   
324      Ice Cream Pad                                           Unknown   

        course          province     region  
0    main dish  Ubon Ratchathani  Northeast  
1    main dish           Bangkok    Central  
2    main dish           Bangkok    Central  
3    main dish         Ayutthaya    Central  
4    main dish           Bangkok    Central  
..         ...               ...        ...  
319    dessert           Unknown    Unknown  
320    dessert           Unknown    Unknown  
321    dessert           Unknown    Unknown  
323    dessert           Unknown    Unknown  
324    dessert           Unknown    Unknown  

[324 rows x 5 columns]

In [5]:
import pandas as pd

In [12]:
filePath = 'thai_dishes_filled.csv'

In [13]:
df = pd.read_csv(filePath)

Count of unknown ingredients

In [14]:
unknown_count = df['ingredients'].str.contains('Unknown', case=False, na=False).sum()
print("Number of 'Unknown' rows:", unknown_count)

Number of 'Unknown' rows: 24


In [15]:
df[df['ingredients'].str.contains('Unknown', case=False, na=False)]

en_name         th_name ingredients     course province  \
61                Sa Uh          สะเออะ     Unknown  main dish  Unknown   
71              Karaang           กะราง     Unknown  main dish  Unknown   
96            Kaeng Ron         แกงร้อน     Unknown  main dish  Unknown   
103       Kaeng Sap Nok        แกงสับนก     Unknown  main dish  Unknown   
105          Kaeng Hong          แกงหอง     Unknown  main dish  Unknown   
106     Kaeng Ngao Ngod     แกงเหงาหงอด     Unknown  main dish  Unknown   
121          Cho Hraeng         จอแหร้ง     Unknown  main dish  Unknown   
126              San Mo           แซนโม     Unknown  main dish  Unknown   
140       Nua Nang Phao     เนื้อนางเพา     Unknown  main dish  Unknown   
141              Bo Toa          บอเตาะ     Unknown  main dish  Unknown   
142             Pla Kok           ปลาคก     Unknown  main dish  Unknown   
155    Yam Kai Yang Tao  ยำไก่อย่างเต่า     Unknown  main dish  Unknown   
157           Yam Sanat         ยำสะนัด     Unknown  main dish  Unknown   
158         La We Kadam       ละแวกะดาม     Unknown  main dish  Unknown   
167                  Oo             อู๋     Unknown  main dish  Unknown   
169          Gruay Thip       กรวยทิพย์     Unknown      snack  Unknown   
175      Karawek Sod Si     การเวกสอดสี     Unknown      snack  Unknown   
177              Ko Kae          โก๋แก่     Unknown      snack  Unknown   
178        Kai Sam Yang     ไก่สามอย่าง     Unknown  main dish  Unknown   
186  Kanom Puang Pratad    ขนมพวงประทัด     Unknown      snack  Unknown   
195         Khao Ho Tod      ข้าวห่อทอด     Unknown      snack  Unknown   
205            Dok Kung         ดอกกุ้ง     Unknown      snack  Unknown   
206             Dok Kai          ดอกไข่     Unknown      snack  Unknown   
223               Srong            สร่ง     Unknown      snack  Unknown   

      region  
61   Unknown  
71   Unknown  
96   Unknown  
103  Unknown  
105  Unknown  
106  Unknown  
121  Unknown  
126  Unknown  
140  Unknown  
141  Unknown  
142  Unknown  
155  Unknown  
157  Unknown  
158  Unknown  
167  Unknown  
169  Unknown  
175  Unknown  
177  Unknown  
178  Unknown  
186  Unknown  
195  Unknown  
205  Unknown  
206  Unknown  
223  Unknown

Update Ingredients Unknown rows

In [16]:
ingredients_update = {
    "Sa Uh": "fermented fish, herbs, chili, lime, fish sauce",
    "Karaang": "pork or beef, garlic, herbs, spices",
    "Kaeng Ron": "coconut milk, noodles, pork or shrimp, shallots, garlic, palm sugar, fish sauce",
    "Kaeng Sap Nok": "minced small fish, red curry paste, local greens, kaffir lime, basil",
    "Kaeng Hong": "fish or meat, herbs, spices, coconut or clear broth",
    "Kaeng Ngao Ngod": "fish, dried chilies, garlic, shallot, galangal, kaffir lime, tamarind, shrimp paste",
    "Cho Hraeng": "prawns or seafood, coconut milk, lemongrass, shallots, turmeric, shrimp paste, tamarind, palm sugar",
    "San Mo": "braised beef or pork, soy sauce, cardamom, clove, cinnamon, shallot, pepper",
    "Nua Nang Phao": "beef or pork, coconut cream, red curry paste, fish sauce, palm sugar",
    "Bo Toa": "minced fish or shrimp, grated coconut, garlic, shallot, kaffir lime, egg, seasonings",
    "Pla Kok": "small fish, pickled greens, lemongrass, garlic, dried chili, salt, soy sauce",
    "Yam Kai Yang Tao": "roasted chicken, chicken offal, coconut milk, chili paste, galangal, lemongrass, lime, fish sauce, sugar",
    "Yam Sanat": "pork or shrimp, dried chilies, garlic, shallots, long beans, banana blossom, roasted rice, fish sauce",
    "La We Kadam": "freshwater crab, garlic, galangal, lemongrass, shrimp paste, chilies, tamarind, fish sauce, herbs",
    "Oo": "beef or fish, chilies, lemongrass, kaffir lime, shallot, garlic, fish sauce",
    "Gruay Thip": "batter of flour, coconut milk, egg, fried into cones with curry or sweet filling",
    "Karawek Sod Si": "minced pork, minced shrimp, Chinese sausage, egg, rice flour, seasoning",
    "Ko Kae": "peanuts, flour, sugar, coconut flavor, oil, salt",
    "Kai Sam Yang": "dried shrimp, peanuts, lemongrass, shallots, ginger, chili, lime, fish sauce",
    "Kanom Puang Pratad": "spring roll skin, minced shrimp and pork, garlic, shallot, egg, deep-fried",
    "Khao Ho Tod": "sticky rice parcels, pork or shrimp filling, coconut, deep-fried",
    "Dok Kung": "bread or dough base, minced shrimp, garlic, fish sauce, seasoning, fried",
    "Dok Kai": "flour, egg, coconut, shaped into flower form, fried or baked",
    "Srong": "minced pork or chicken, garlic, pepper, seasoning, wrapped in noodles, deep-fried"
}

In [17]:
df['ingredients'] = df.apply(
    lambda row: ingredients_update.get(row['en_name'], row['ingredients']),
    axis=1
)

In [18]:
df.to_csv('thai_dish_filled_complete.csv', index=False)
print("Updated CSV saved as thai_dish_filled_complete.csv")

Updated CSV saved as thai_dish_filled_complete.csv


In [19]:
unknown_count = df['ingredients'].str.contains('Unknown', case=False, na=False).sum()
print("Number of 'Unknown' rows:", unknown_count)

Number of 'Unknown' rows: 0


All ingredients require for dishes are filles.

Unknown data has filled in Region and Province too !!!! Have to clean !!!!!

First: need to count and view those data rows.

In [20]:
unknown_province = (df['province'] == 'Unknown').sum()
unknown_region = (df['region'] == 'Unknown').sum()

print("Unknown in 'province':", unknown_province)
print("Unknown in 'region':", unknown_region)

Unknown in 'province': 25
Unknown in 'region': 25


Total 25Unknown Province and 25Unknown Region Unknown. I have fill these.

In [22]:
df[(df['province'] == 'Unknown') | (df['region'] == 'Unknown')]

en_name         th_name  \
13            Kai Katet       ไก่กะเต็ด   
61                Sa Uh          สะเออะ   
71              Karaang           กะราง   
96            Kaeng Ron         แกงร้อน   
103       Kaeng Sap Nok        แกงสับนก   
105          Kaeng Hong          แกงหอง   
106     Kaeng Ngao Ngod     แกงเหงาหงอด   
121          Cho Hraeng         จอแหร้ง   
126              San Mo           แซนโม   
140       Nua Nang Phao     เนื้อนางเพา   
141              Bo Toa          บอเตาะ   
142             Pla Kok           ปลาคก   
155    Yam Kai Yang Tao  ยำไก่อย่างเต่า   
157           Yam Sanat         ยำสะนัด   
158         La We Kadam       ละแวกะดาม   
167                  Oo             อู๋   
169          Gruay Thip       กรวยทิพย์   
175      Karawek Sod Si     การเวกสอดสี   
177              Ko Kae          โก๋แก่   
178        Kai Sam Yang     ไก่สามอย่าง   
186  Kanom Puang Pratad    ขนมพวงประทัด   
195         Khao Ho Tod      ข้าวห่อทอด   
205            Dok Kung         ดอกกุ้ง   
206             Dok Kai          ดอกไข่   
223               Srong            สร่ง   

                                           ingredients     course province  \
13                     Chicken+spices+soy sauce+garlic  main dish  Unknown   
61      fermented fish, herbs, chili, lime, fish sauce  main dish  Unknown   
71                 pork or beef, garlic, herbs, spices  main dish  Unknown   
96   coconut milk, noodles, pork or shrimp, shallot...  main dish  Unknown   
103  minced small fish, red curry paste, local gree...  main dish  Unknown   
105  fish or meat, herbs, spices, coconut or clear ...  main dish  Unknown   
106  fish, dried chilies, garlic, shallot, galangal...  main dish  Unknown   
121  prawns or seafood, coconut milk, lemongrass, s...  main dish  Unknown   
126  braised beef or pork, soy sauce, cardamom, clo...  main dish  Unknown   
140  beef or pork, coconut cream, red curry paste, ...  main dish  Unknown   
141  minced fish or shrimp, grated coconut, garlic,...  main dish  Unknown   
142  small fish, pickled greens, lemongrass, garlic...  main dish  Unknown   
155  roasted chicken, chicken offal, coconut milk, ...  main dish  Unknown   
157  pork or shrimp, dried chilies, garlic, shallot...  main dish  Unknown   
158  freshwater crab, garlic, galangal, lemongrass,...  main dish  Unknown   
167  beef or fish, chilies, lemongrass, kaffir lime...  main dish  Unknown   
169  batter of flour, coconut milk, egg, fried into...      snack  Unknown   
175  minced pork, minced shrimp, Chinese sausage, e...      snack  Unknown   
177   peanuts, flour, sugar, coconut flavor, oil, salt      snack  Unknown   
178  dried shrimp, peanuts, lemongrass, shallots, g...  main dish  Unknown   
186  spring roll skin, minced shrimp and pork, garl...      snack  Unknown   
195  sticky rice parcels, pork or shrimp filling, c...      snack  Unknown   
205  bread or dough base, minced shrimp, garlic, fi...      snack  Unknown   
206  flour, egg, coconut, shaped into flower form, ...      snack  Unknown   
223  minced pork or chicken, garlic, pepper, season...      snack  Unknown   

      region  
13   Unknown  
61   Unknown  
71   Unknown  
96   Unknown  
103  Unknown  
105  Unknown  
106  Unknown  
121  Unknown  
126  Unknown  
140  Unknown  
141  Unknown  
142  Unknown  
155  Unknown  
157  Unknown  
158  Unknown  
167  Unknown  
169  Unknown  
175  Unknown  
177  Unknown  
178  Unknown  
186  Unknown  
195  Unknown  
205  Unknown  
206  Unknown  
223  Unknown

Using iloc this time

In [23]:
df.loc[df['en_name'] == 'Kai Katet', ['province', 'region']] = ['Chiang Mai', 'Northern']
df.loc[df['en_name'] == 'Sa Uh', ['province', 'region']] = ['Chiang Mai', 'Northern']
df.loc[df['en_name'] == 'Karaang', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Kaeng Ron', ['province', 'region']] = ['Chiang Mai', 'Northern']
df.loc[df['en_name'] == 'Kaeng Sap Nok', ['province', 'region']] = ['Ayutthaya', 'Central']
df.loc[df['en_name'] == 'Kaeng Hong', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Kaeng Ngao Ngod', ['province', 'region']] = ['Ayutthaya', 'Central']
df.loc[df['en_name'] == 'Cho Hraeng', ['province', 'region']] = ['Phang Nga', 'Southern']
df.loc[df['en_name'] == 'San Mo', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Nua Nang Phao', ['province', 'region']] = ['Phuket', 'Southern']
df.loc[df['en_name'] == 'Bo Toa', ['province', 'region']] = ['Songkhla', 'Southern']
df.loc[df['en_name'] == 'Pla Kok', ['province', 'region']] = ['Chonburi', 'Eastern']
df.loc[df['en_name'] == 'Yam Kai Yang Tao', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Yam Sanat', ['province', 'region']] = ['Chiang Mai', 'Northern']
df.loc[df['en_name'] == 'La We Kadam', ['province', 'region']] = ['Nakhon Ratchasima', 'Northeast']
df.loc[df['en_name'] == 'Oo', ['province', 'region']] = ['Khon Kaen', 'Northeast']
df.loc[df['en_name'] == 'Gruay Thip', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Karawek Sod Si', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Ko Kae', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Kai Sam Yang', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Kanom Puang Pratad', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Khao Ho Tod', ['province', 'region']] = ['Chiang Mai', 'Northern']
df.loc[df['en_name'] == 'Dok Kung', ['province', 'region']] = ['Pattani', 'Southern']
df.loc[df['en_name'] == 'Dok Kai', ['province', 'region']] = ['Bangkok', 'Central']
df.loc[df['en_name'] == 'Srong', ['province', 'region']] = ['Bangkok', 'Central']

Add Notes column

In [24]:
df['Note'] = df['en_name'].map({
    'Kai Katet': 'Possibly a historical Northern Thai chicken dish; region and province inferred from old recipes, not verified.',
    'Sa Uh': 'Traditional Northern fermented fish dish; inferred from Lanna cuisine, uncertain documentation.',
    'Karaang': 'Central Thai-style curry; origin uncertain, referenced in old culinary texts.',
    'Kaeng Ron': 'Historical spicy Northern soup; origin uncertain.',
    'Kaeng Sap Nok': 'Classic Central curry from old Ayutthaya recipes; possibly historical.',
    'Kaeng Hong': 'Old Central Thai curry; region inferred from literature.',
    'Kaeng Ngao Ngod': 'Historical Central curry; uncertain provenance.',
    'Cho Hraeng': 'Southern seafood curry; uncertain provenance from old cookbooks.',
    'San Mo': 'Likely Central Thai braised dish; not widely known, possibly historical.',
    'Nua Nang Phao': 'Southern beef curry; uncertain but appears in coastal cuisine texts.',
    'Bo Toa': 'Southern minced seafood dish; found in historical Songkhla records.',
    'Pla Kok': 'Eastern fish curry; uncertain but likely coastal traditional dish.',
    'Yam Kai Yang Tao': 'Central-style chicken salad; region inferred from Central Thai recipes.',
    'Yam Sanat': 'Northern Thai-style salad; historical but undocumented.',
    'La We Kadam': 'Northeastern crab dish; possibly historical, uncertain source.',
    'Oo': 'Traditional Isan soup; uncertain documentation.',
    'Gruay Thip': 'Central Thai dessert; inferred from dessert collections, uncertain author.',
    'Karawek Sod Si': 'Historical royal snack; Central Thai origin.',
    'Ko Kae': 'Modern snack brand adapted from traditional recipe; not historical.',
    'Kai Sam Yang': 'Central Thai chicken mix dish; uncertain documentation.',
    'Kanom Puang Pratad': 'Central fried snack; possible historical version.',
    'Khao Ho Tod': 'Northern-style fried rice parcels; uncertain provenance.',
    'Dok Kung': 'Southern shrimp-based snack; old regional reference.',
    'Dok Kai': 'Traditional flower-shaped Central Thai dessert; uncertain author.',
    'Srong': 'Historical royal snack from Central region; documented in old texts.'
}).fillna('Region and province inferred from old sources; no verified documentation.')

In [25]:
df

en_name            th_name  \
0                  Koi               ก้อย   
1            Kuai Chap            กวยจั๊บ   
2            Kuai Tiao         ก๋วยเตี๋ยว   
3      Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4    Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
..                 ...                ...   
320              A Kao             อาเกาะ   
321              A Law           อาละหว่า   
322              A Law              อาลัว   
323              O Aew            โอ้เอ๋ว   
324      Ice Cream Pad           ไอติมผัด   

                                          ingredients     course  \
0              Beef+lime juice+fish sauce+chili+herbs  main dish   
1            Pork belly+rice noodles+soy sauce+garlic  main dish   
2    Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3              Beef+noodles+curry powder+coconut milk  main dish   
4                    Chicken+noodles+soy sauce+garlic  main dish   
..                                                ...        ...   
320           Eggs+Coconut Milk+Palm Sugar+Rice Flour    dessert   
321               Alua candy+flour+coconut milk+sugar    dessert   
322               Alua candy+flour+coconut milk+sugar    dessert   
323                Banana‑seed jelly+shaved ice+syrup    dessert   
324                Rolled ice cream+milk base+mix‑ins    dessert   

             province     region  \
0    Ubon Ratchathani  Northeast   
1             Bangkok    Central   
2             Bangkok    Central   
3           Ayutthaya    Central   
4             Bangkok    Central   
..                ...        ...   
320           Pattani      South   
321       Phetchaburi    Central   
322       Phetchaburi    Central   
323            Phuket      South   
324           Various    Various   

                                                  Note  
0    Region and province inferred from old sources;...  
1    Region and province inferred from old sources;...  
2    Region and province inferred from old sources;...  
3    Region and province inferred from old sources;...  
4    Region and province inferred from old sources;...  
..                                                 ...  
320  Region and province inferred from old sources;...  
321  Region and province inferred from old sources;...  
322  Region and province inferred from old sources;...  
323  Region and province inferred from old sources;...  
324  Region and province inferred from old sources;...  

[325 rows x 7 columns]

Have to save it

In [26]:
df.to_csv('thai_dish_filled_complete.csv', index=False)

In [27]:
df

en_name            th_name  \
0                  Koi               ก้อย   
1            Kuai Chap            กวยจั๊บ   
2            Kuai Tiao         ก๋วยเตี๋ยว   
3      Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4    Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
..                 ...                ...   
320              A Kao             อาเกาะ   
321              A Law           อาละหว่า   
322              A Law              อาลัว   
323              O Aew            โอ้เอ๋ว   
324      Ice Cream Pad           ไอติมผัด   

                                          ingredients     course  \
0              Beef+lime juice+fish sauce+chili+herbs  main dish   
1            Pork belly+rice noodles+soy sauce+garlic  main dish   
2    Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3              Beef+noodles+curry powder+coconut milk  main dish   
4                    Chicken+noodles+soy sauce+garlic  main dish   
..                                                ...        ...   
320           Eggs+Coconut Milk+Palm Sugar+Rice Flour    dessert   
321               Alua candy+flour+coconut milk+sugar    dessert   
322               Alua candy+flour+coconut milk+sugar    dessert   
323                Banana‑seed jelly+shaved ice+syrup    dessert   
324                Rolled ice cream+milk base+mix‑ins    dessert   

             province     region  \
0    Ubon Ratchathani  Northeast   
1             Bangkok    Central   
2             Bangkok    Central   
3           Ayutthaya    Central   
4             Bangkok    Central   
..                ...        ...   
320           Pattani      South   
321       Phetchaburi    Central   
322       Phetchaburi    Central   
323            Phuket      South   
324           Various    Various   

                                                  Note  
0    Region and province inferred from old sources;...  
1    Region and province inferred from old sources;...  
2    Region and province inferred from old sources;...  
3    Region and province inferred from old sources;...  
4    Region and province inferred from old sources;...  
..                                                 ...  
320  Region and province inferred from old sources;...  
321  Region and province inferred from old sources;...  
322  Region and province inferred from old sources;...  
323  Region and province inferred from old sources;...  
324  Region and province inferred from old sources;...  

[325 rows x 7 columns]

In [39]:
df = pd.read_csv('thai_dish_filled_complete.csv')

In [40]:
df

en_name            th_name  \
0                  Koi               ก้อย   
1            Kuai Chap            กวยจั๊บ   
2            Kuai Tiao         ก๋วยเตี๋ยว   
3      Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4    Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
..                 ...                ...   
320              A Kao             อาเกาะ   
321              A Law           อาละหว่า   
322              A Law              อาลัว   
323              O Aew            โอ้เอ๋ว   
324      Ice Cream Pad           ไอติมผัด   

                                          ingredients     course  \
0              Beef+lime juice+fish sauce+chili+herbs  main dish   
1            Pork belly+rice noodles+soy sauce+garlic  main dish   
2    Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3              Beef+noodles+curry powder+coconut milk  main dish   
4                    Chicken+noodles+soy sauce+garlic  main dish   
..                                                ...        ...   
320           Eggs+Coconut Milk+Palm Sugar+Rice Flour    dessert   
321               Alua candy+flour+coconut milk+sugar    dessert   
322               Alua candy+flour+coconut milk+sugar    dessert   
323                Banana‑seed jelly+shaved ice+syrup    dessert   
324                Rolled ice cream+milk base+mix‑ins    dessert   

             province     region  \
0    Ubon Ratchathani  Northeast   
1             Bangkok    Central   
2             Bangkok    Central   
3           Ayutthaya    Central   
4             Bangkok    Central   
..                ...        ...   
320           Pattani      South   
321       Phetchaburi    Central   
322       Phetchaburi    Central   
323            Phuket      South   
324           Various    Various   

                                                  Note  
0    Region and province inferred from old sources;...  
1    Region and province inferred from old sources;...  
2    Region and province inferred from old sources;...  
3    Region and province inferred from old sources;...  
4    Region and province inferred from old sources;...  
..                                                 ...  
320  Region and province inferred from old sources;...  
321  Region and province inferred from old sources;...  
322  Region and province inferred from old sources;...  
323  Region and province inferred from old sources;...  
324  Region and province inferred from old sources;...  

[325 rows x 7 columns]

In [41]:
df = df.drop(columns=['Note'])

In [42]:
df

en_name            th_name  \
0                  Koi               ก้อย   
1            Kuai Chap            กวยจั๊บ   
2            Kuai Tiao         ก๋วยเตี๋ยว   
3      Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4    Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
..                 ...                ...   
320              A Kao             อาเกาะ   
321              A Law           อาละหว่า   
322              A Law              อาลัว   
323              O Aew            โอ้เอ๋ว   
324      Ice Cream Pad           ไอติมผัด   

                                          ingredients     course  \
0              Beef+lime juice+fish sauce+chili+herbs  main dish   
1            Pork belly+rice noodles+soy sauce+garlic  main dish   
2    Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3              Beef+noodles+curry powder+coconut milk  main dish   
4                    Chicken+noodles+soy sauce+garlic  main dish   
..                                                ...        ...   
320           Eggs+Coconut Milk+Palm Sugar+Rice Flour    dessert   
321               Alua candy+flour+coconut milk+sugar    dessert   
322               Alua candy+flour+coconut milk+sugar    dessert   
323                Banana‑seed jelly+shaved ice+syrup    dessert   
324                Rolled ice cream+milk base+mix‑ins    dessert   

             province     region  
0    Ubon Ratchathani  Northeast  
1             Bangkok    Central  
2             Bangkok    Central  
3           Ayutthaya    Central  
4             Bangkok    Central  
..                ...        ...  
320           Pattani      South  
321       Phetchaburi    Central  
322       Phetchaburi    Central  
323            Phuket      South  
324           Various    Various  

[325 rows x 6 columns]

In [43]:
unknown_province = (df['province'] == 'Unknown').sum()
unknown_region = (df['region'] == 'Unknown').sum()

print("Unknown in 'province':", unknown_province)
print("Unknown in 'region':", unknown_region)

Unknown in 'province': 0
Unknown in 'region': 0


In [44]:
df.to_csv('thai_dish_filled_complete.csv', index=False)

In [46]:
various_province = (df['province'] == 'Various').sum()
various_region = (df['region'] == 'Various').sum()

print("Various in 'province':", various_province)
print("Various in 'region':", various_region)

Various in 'province': 148
Various in 'region': 148


All Unknown data in Province and Region are filled but there is still left 'Various' that we don't understand. And we have to fix this!!!!!!!!

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('thai_dish_filled_complete.csv')

In [3]:
duplicate_count = df.duplicated().sum()
print("Total duplicate rows:", duplicate_count)

Total duplicate rows: 0


In [4]:
df.tail()

en_name   th_name  \
319         Ang Ku    อังกู๊   
320          A Kao    อาเกาะ   
321          A Law  อาละหว่า   
322          O Aew   โอ้เอ๋ว   
323  Ice Cream Pad  ไอติมผัด   

                                           ingredients   course     province  \
319  Glutinous rice cakes+mung‑bean filling (Hokkie...  dessert       Phuket   
320            Eggs+Coconut Milk+Palm Sugar+Rice Flour  dessert      Pattani   
321                Alua candy+flour+coconut milk+sugar  dessert  Phetchaburi   
322                 Banana‑seed jelly+shaved ice+syrup  dessert       Phuket   
323                 Rolled ice cream+milk base+mix‑ins  dessert      Various   

      region  
319    South  
320    South  
321  Central  
322    South  
323  Various

In [5]:
df.tail(10)

en_name        th_name  \
314       Lot Chong        ลอดช่อง   
315        Luk Chup         ลูกชุบ   
316        Som Chun         ส้มฉุน   
317   Sankhaya Kuan   สังขยา (กวน)   
318  Sankhaya Nueng  สังขยา (นึ่ง)   
319          Ang Ku         อังกู๊   
320           A Kao         อาเกาะ   
321           A Law       อาละหว่า   
322           O Aew        โอ้เอ๋ว   
323   Ice Cream Pad       ไอติมผัด   

                                           ingredients   course     province  \
314                          Flour+pandan+coconut milk  dessert      Various   
315                      Mung beans+coconut milk+sugar  dessert      Various   
316                             Herbs+fruit+fish sauce    salad      Various   
317                             Flour+coconut milk+egg  dessert      Various   
318                             Flour+coconut milk+egg  dessert      Various   
319  Glutinous rice cakes+mung‑bean filling (Hokkie...  dessert       Phuket   
320            Eggs+Coconut Milk+Palm Sugar+Rice Flour  dessert      Pattani   
321                Alua candy+flour+coconut milk+sugar  dessert  Phetchaburi   
322                 Banana‑seed jelly+shaved ice+syrup  dessert       Phuket   
323                 Rolled ice cream+milk base+mix‑ins  dessert      Various   

      region  
314  Various  
315  Various  
316  Various  
317  Various  
318  Various  
319    South  
320    South  
321  Central  
322    South  
323  Various